exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/scan/framework/AbstractScanFramework.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.physical.impl.scan.framework;
+
+import java.util.List;
+
+import org.apache.drill.common.expression.SchemaPath;
+import org.apache.drill.common.types.TypeProtos.MajorType;
+import org.apache.drill.exec.ops.OperatorContext;
+import org.apache.drill.exec.physical.impl.scan.ScanOperatorEvents;
+import org.apache.drill.exec.physical.impl.scan.project.ScanSchemaOrchestrator;
+
+/**
+ * Basic scan framework for a "managed" reader which uses the scan schema
+ * mechanisms encapsulated in the scan schema orchestrator. Handles binding
+ * scan events to the scan orchestrator so that the scan schema is evolved
+ * as the scan progresses. Subclasses are responsible for creating the actual
+ * reader, which requires a framework-specific schema negotiator to be passed
+ * to the reader.
+ * <p>
+ * This framework is a bridge between operator logic and the scan projection
+ * internals. It gathers scan-specific options, then sets
```

Comment:
` It gathers scan-specific options, then sets then on the scan orchestrator at the right time.`
Looks like we have extra `then` here.